permissions_validators/src/public_blockchain/transfer.rs

```diff
@@ -103,3 +115,133 @@ impl<W: WorldTrait> IsGrantAllowed<W> for GrantMyAssetAccess {
+
+impl_from_item_for_instruction_validator_box!(ExecutionCountFitsInLimit);
+
+impl<W: WorldTrait> IsAllowed<W, Instruction> for ExecutionCountFitsInLimit {
+    fn check(
+        &self,
+        authority: &AccountId,
+        instruction: &Instruction,
+        wsv: &WorldStateView<W>,
+    ) -> Result<(), DenialReason> {
+        if !matches!(instruction, Instruction::Transfer(_)) {
+            return Ok(());
+        };
+
+        let params = match retrieve_permission_params(wsv, authority)? {
+            Some(params) => params,
+            None => return Ok(()),
+        };
+
+        let period = retrieve_period(&params)?;
+        let count = retrieve_count(&params)?;
+
+        if count_executions(wsv, authority, period) >= count {
+            return Err(DenialReason::from(
+                "Transfer transaction limit for current period is exceed",
+            ));
+        }
+        Ok(())
+    }
+}
+
+/// Retrieve permission parameters for `ExecutionCountFitsInLimit` validator.
+/// Returns Some if there is a suitable token and None if not
+///
+/// # Errors
+/// - Account doesn't exist
+fn retrieve_permission_params<W: WorldTrait>(
+    wsv: &WorldStateView<W>,
+    authority: &AccountId,
+) -> Result<Option<BTreeMap<Name, Value>>, DenialReason> {
+    wsv.map_account(authority, |account| {
+        wsv.account_permission_tokens(account)
+            .iter()
+            .filter(|token| token.name == *CAN_TRANSFER_ONLY_FIXED_NUMBER_OF_TIMES_PER_PERIOD)
+            .map(|token| token.params.clone())
+            .next()
+    })
+    .map_err(|e| e.to_string())
+}
+
+/// Retrieve period from `params`
+///
+/// # Errors
+/// - There is no period parameter
+/// - Period has wrong value type
+fn retrieve_period(params: &BTreeMap<Name, Value>) -> Result<Duration, DenialReason> {
+    match params
+        .get(&*PERIOD_PARAM_NAME)
+        .ok_or_else(|| format!("Expected `{}` parameter", *PERIOD_PARAM_NAME))?
+    {
+        Value::U128(period) => Ok(Duration::from_millis(
+            u64::try_from(*period).map_err(|e| e.to_string())?,
+        )),
+        _ => Err(format!(
+            "`{}` parameter has wrong value type. Expected `u128`",
+            *PERIOD_PARAM_NAME
+        )),
+    }
+}
+
+/// Retrieve count from `params`
+///
+/// # Errors
+/// - There is no count parameter
+/// - Count has wrong value type
+fn retrieve_count(params: &BTreeMap<Name, Value>) -> Result<u32, DenialReason> {
+    match params
+        .get(&*COUNT_PARAM_NAME)
+        .ok_or_else(|| format!("Expected `{}` parameter", *COUNT_PARAM_NAME))?
+    {
+        Value::U32(count) => Ok(*count),
+        _ => Err(format!(
+            "`{}` parameter has wrong value type. Expected `u32`",
+            *COUNT_PARAM_NAME
+        )),
+    }
+}
+
+/// Find number of `Transfer` execution count which happened in last `period`
+fn count_executions<W: WorldTrait>(
+    wsv: &WorldStateView<W>,
+    authority: &AccountId,
+    period: Duration,
+) -> u32 {
```

Comment:
It makes more sense that you return `usize` here and handle conversion at the call site if needed